### PR TITLE
Add istioctl wait to reachability tests

### PR DESCRIPTION
**Pull request by howardjohn**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
Test time from 135s -> 85s in one example I looked at

File: tests/integration/security/util/reachability/context.go [hunk 1]
```diff
@@ -155,16 +157,15 @@ func (rc *Context) Run(testCases []TestCase) {
 				// TODO(https://github.com/istio/istio/issues/20460) We shouldn't need a retry loop
 				return rc.ctx.Config().ApplyYAML(c.Namespace.Name(), policyYAML)
 			})
+			ctx.Logf("[%s] [%v] Wait for config propagate to endpoints...", testName, time.Now())
+			if err := ik.WaitForConfigs(c.Namespace.Name(), policyYAML); err != nil {
+				ctx.Fatal(err)
```
Comment (incfly) [inline tests/integration/security/util/reachability/context.go:162]: wraup with Istioctl wait fails?

File: tests/integration/security/util/reachability/context.go [hunk 2]
```diff
@@ -155,16 +157,15 @@ func (rc *Context) Run(testCases []TestCase) {
 				// TODO(https://github.com/istio/istio/issues/20460) We shouldn't need a retry loop
 				return rc.ctx.Config().ApplyYAML(c.Namespace.Name(), policyYAML)
 			})
+			ctx.Logf("[%s] [%v] Wait for config propagate to endpoints...", testName, time.Now())
+			if err := ik.WaitForConfigs(c.Namespace.Name(), policyYAML); err != nil {
```
Comment (incfly) [inline tests/integration/security/util/reachability/context.go:161]: If I understand correctly, there's no ordering the apply is happening before Wait. and wait is robust to such sequencing?

**Pull request merged**